enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
```

Comment:
- [ ] enumerables #my_each returns an Enumerator if no block is given
- [ ] enumerables #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- [ ] enumerables #my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self
- [ ]